test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1

```diff
@@ -0,0 +1,345 @@
+
+        $Var1 = "Hello"
+        $Var2 = "Goodbye"
+        $Var3 = 102
+        $Var4 = 1..5
+        $global:GVar1 = "GlobalVar"
+        $job = Start-ThreadJob -ScriptBlock {
+            Write-Output $using:Var1
+            Write-Output $using:Var2
+            Write-Output $using:Var3
+            Write-Output ($using:Var4)[1]
+            Write-Output @(,$using:Var4)
+            Write-Output $using:GVar1
+        }
+
+        $results = $job | Receive-Job -Wait
+        $results[0] | Should Be $Var1
+        $results[1] | Should Be $Var2
+        $results[2] | Should Be $Var3
+        $results[3] | Should Be 2
+        $results[4] | Should Be $Var4
+        $results[5] | Should Be $global:GVar1
+    }
+
+    It 'ThreadJob with ScriptBlock and Using variables and argument list' {
+
+        $Var1 = "Hello"
+        $Var2 = 52
+        $job = Start-ThreadJob -ScriptBlock {
+            param ([string] $param1)
+
+            "$using:Var1 $param1 $using:Var2"
+        } -ArgumentList "There"
+
+        $results = $job | Receive-Job -Wait
+        $results | Should Be "Hello There 52"
+    }
+
+    It 'ThreadJob with ScriptFile' {
+
+        $job = Start-ThreadJob -FilePath $scriptFilePath1
+        $results = $job | Receive-Job -Wait
+        $results.Count | Should be 10
+        $results[9] | Should be "Hello 9"
+    }
+
+    It 'ThreadJob with ScriptFile and Initialization script' {
+
+        $job = Start-ThreadJob -FilePath $scriptFilePath1 -Initialization { "Goodbye" }
+        $results = $job | Receive-Job -Wait
+        $results.Count | Should be 11
+        $results[0] | Should be "Goodbye"
+    }
+
+    It 'ThreadJob with ScriptFile and Argument list' {
+
+        $job = Start-ThreadJob -FilePath $scriptFilePath2 -ArgumentList @("Hello","Goodbye")
+        $results = $job | Receive-Job -Wait
+        $results[0] | Should be "Hello"
+        $results[1] | Should be "Goodbye"
+    }
+
+    It 'ThreadJob with ScriptFile and piped input' {
+
+        $job = "Hello","Goodbye" | Start-ThreadJob -FilePath $scriptFilePath3
+        $results = $job | Receive-Job -Wait
+        $results[0] | Should be "Hello"
+        $results[1] | Should be "Goodbye"
+    }
+
+    It 'ThreadJob with ScriptFile and Using variables' {
+
+        $Var1 = "Hello!"
+        $Array1 = 1..10
+
+        $job = Start-ThreadJob -FilePath $scriptFilePath4
+        $results = $job | Receive-Job -Wait
+        $results[0] | Should be $Var1
+        $results[1] | Should be 3
+        $results[2] | Should be $Array1
+    }
+
+    It 'ThreadJob with ScriptFile and Using variables with argument list' {
+
+        $Var1 = "There"
+        $Var2 = 60
+        $job = Start-ThreadJob -FilePath $scriptFilePath5 -ArgumentList "Hello"
+        $results = $job | Receive-Job -Wait
+        $results | Should Be "Hello There 60"
+    }
+
+    It 'ThreadJob with terminating error' {
+
+        $job = Start-ThreadJob -ScriptBlock { throw "MyError!" }
```

Comment:
Can we also add a test for a non-terminating error where some objects are returned and 1 non-terminating error is returned?

Reply:
This is outside the scope of this change.  Please create an Issue for adding more threadjob tests.